OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1014,6 +1014,68 @@ internal static class DreamProcNativeRoot {
             return new DreamValue(obj.ObjectDefinition.HasProc(procName) ? 1 : 0);
         }
 
+        [DreamProc("hearers")]
+        [DreamProcParameter("Depth", Type = DreamValueTypeFlag.Float)]
+        [DreamProcParameter("Center", Type = DreamValueTypeFlag.DreamObject)]
+        public static DreamValue NativeProc_hearers(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) { //TODO: Change depending on center
+            DreamValue depthValue = new DreamValue(5);
+            DreamObjectAtom? center = null;
+
+            //Arguments are optional and can be passed in any order
+            if (bundle.Arguments.Length > 0) {
+                DreamValue firstArgument = bundle.GetArgument(0, "Depth");
+
+                if (firstArgument.TryGetValueAsDreamObject(out center)) {
+                    if (bundle.Arguments.Length > 1) {
+                        depthValue = bundle.GetArgument(1, "Center");
+                    }
+                } else {
+                    depthValue = firstArgument;
+
+                    if (bundle.Arguments.Length > 1) {
+                        bundle.GetArgument(1, "Center").TryGetValueAsDreamObject(out center);
+                    }
+                }
+            }
+
+            center ??= usr as DreamObjectAtom;
+
+            DreamList hear = bundle.ObjectTree.CreateList();
+            if (center == null)
+                return new(hear);
+
+            var centerPos = bundle.AtomManager.GetAtomPosition(center);
+            if (!depthValue.TryGetValueAsInteger(out var depth))
+                depth = 5;
+
+            foreach (var atom in bundle.AtomManager.EnumerateAtoms(bundle.ObjectTree.Mob)) {
+                var mob = (DreamObjectMob)atom;
+
+                if (centerPos.Z == mob.Z && Math.Abs(centerPos.X - mob.X) <= depth && Math.Abs(centerPos.Y - mob.Y) <= depth) {
```

Comment:
Are you doing `in range(5)` and *then* doing `in range(view string)`? Because that doesn't seem right.
It seems like you should instead be gathering the tiles in range, calculating audibility, *then* iterating the mobs to see if they're on an audible tile

Reply:
this is how viewers() implements it. I didn't look too closely but from what I can tell, line 1048 tries to get depth and defaults to 5 if it fails
```cs
            if (!depthValue.TryGetValueAsInteger(out var depth))
                depth = 5;
```